spec/ci_helper_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'spec_helper.rb'
+require 'webmock/rspec'
+
+shared_examples "a CI provider" do 
+    subject { model }
+    it { is_expected.to respond_to(:login) }
+    it { is_expected.to respond_to(:organization) }
+    it { is_expected.to respond_to(:repository) }
+    it { is_expected.to respond_to(:trigger_job) }
+end
+
+describe Fastlane::Helper::CircleCIHelper do
+    it_behaves_like "a CI provider" do 
+        let(:model) { Fastlane::Helper::CircleCIHelper.new(login: "my_circleci_token", repository: "test_repo") }
+    end
+
+    context "initialization" do 
+        describe Fastlane::Helper::CircleCIHelper.new(login: "my_circleci_token", repository: "test_repo") do 
+            it { is_expected.to have_attributes(:organization => "wordpress-mobile", "repository" => "test_repo", "command_uri" => URI.parse("https://circleci.com/api/v2/project/github/wordpress-mobile/test_repo/pipeline"))}
+        end
```

Comment:
TIL we can provide an _instance_ to `describe` and use it inside `it { … }` block. I always only used module/class names like `describe Fastlane::Helper::CircleCIHelper do` only. 🤯

Reply:
That's pretty cool (and as meta as only Ruby can get ❤️ )